src/librustdoc/doctest/make.rs

```diff
@@ -587,14 +584,9 @@ fn parse_source(
                             }
                         }
                     }
-                    StmtKind::Expr(ref expr) => {
-                        if matches!(expr.kind, ast::ExprKind::Err(_)) {
-                            reset_error_count(&psess);
-                            return Err(());
-                        }
```

Comment:
We no longer need to do that. Due to the `ErrorGuaranteed` in `ExprKind::Err` we *know* that an error diagnostic had to be emitted prior meaning `dcx.has_errors()` *has* to be some (meaning we already bailed out with an `Err(_)` above)!